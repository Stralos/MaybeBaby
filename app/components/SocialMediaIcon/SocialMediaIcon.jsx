/* @flow */
import React from 'react';
import styled from 'styled-components';

type Props = {
  +type?: string
}

const SocialMediaIcon = styled.span`
  height: 24px;
  width: 24px;
  display: inline-block;
  background-color: ${(props: Props) => {
    return props.type;
  }}
`;

export default (props: Props) => <SocialMediaIcon {...props} />;
