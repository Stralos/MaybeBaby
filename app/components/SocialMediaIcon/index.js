/* @flow */
import SocialMediaIcon from './SocialMediaIcon';

export { FACEBOOK, TWITTER, INSTAGRAM } from './SocialMediaIcon';
export default SocialMediaIcon;
