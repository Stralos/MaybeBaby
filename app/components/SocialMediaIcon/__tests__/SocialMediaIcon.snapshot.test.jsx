/* @flow */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import SocialMediaIcon from '../SocialMediaIcon';

describe('SocialMediaIcon Snapshot Testing --', () => {
  it('should render and match snapshot', () => {
    const component = renderer.create(<SocialMediaIcon type={'Test'} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
