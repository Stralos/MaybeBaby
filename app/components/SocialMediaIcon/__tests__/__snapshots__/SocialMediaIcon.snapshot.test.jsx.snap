// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaIcon Snapshot Testing -- should render and match snapshot 1`] = `
.c0 {
  height: 24px;
  width: 24px;
  display: inline-block;
  background-color: Test;
}

<span
  className="c0"
  type="Test"
/>
`;
